desc/basis.py

```diff
@@ -1304,6 +1555,30 @@ def powers(rho, l, dr=0):
     return polyval_vec(coeffs, rho).T
 
 
+@jit
+def chebyshev(r, l, dr=0):
+    """Shifted Chebyshev polynomial.
+
+    Parameters
+    ----------
+    rho : ndarray, shape(N,)
+        radial coordinates to evaluate basis
+    l : ndarray of int, shape(K,)
+        radial mode number(s)
+    dr : int
+        order of derivative (Default = 0)
+
+    Returns
+    -------
+    y : ndarray, shape(N,K)
+        basis function(s) evaluated at specified points
+
+    """
+    x = 2 * r - 1  # shift
+    x, l, dr = map(jnp.asarray, (x, l, dr))
+    return jnp.cos(l * jnp.arccos(x))
```

Comment:
Are there any possible cases where AD would try to differentiate this wrt `r`? IE, doesn't the omnigenity stuff require evaluating stuff at new nodes?

Reply:
I don't think so. In the omnigenity computation, the Chebyshev basis is only used to interpolate the "well shape" on the flux surface of interest. So the transform matrix that calls this function is pre-computed. The Boozer coordinate basis for $|B|$ is the transform matrix with changing nodes that has to be computed on the fly with AD, but that is a double Fourier series.